### DISCUSSION TITLE
Build app on Windows - What is the setup?

### DISCUSSION THREAD

**Discussion by loicngr** [tauri-apps/tauri#1608] [Q&A]
Hello,

I wanted to test a tauri exemple project, but I have a problem when I launch the command `yarn tauri dev`. Should I add a configuration for windows?

```
...
Compiling reqwest v0.11.3
Finished dev [unoptimized + debuginfo] target(s) in 1m 26s
   Running `target\debug\app.exe`
error: could not execute process `target\debug\app.exe` (never executed)

Caused by:
  This version of %1 is not compatible with the version of Windows you're running. Check your computer's system information and then contact the software publisher. (os error 216)
````

More informations about my setup : 
````
F:\perso\tauri-app (master)
λ yarn tauri info
yarn run v1.22.10
$ F:\perso\tauri-app\node_modules\.bin\tauri info
Downloading Tauri CLI
Download Complete
 app:spawn Running "F:\perso\tauri-app\node_modules\@tauri-apps\cli\bin\tauri-cli.exe tauri info" +0ms


Operating System - Windows, version 10.0.19042 X64

Node.js environment
  Node.js - 12.5.0
  @tauri-apps/cli - 1.0.0-beta-rc.2
  @tauri-apps/api - 1.0.0

Global packages
  npm - 6.9.0
  yarn - 1.22.10

Rust environment
  rustc - 1.51.0
  cargo - 1.51.0

App directory structure
/.git
/.github
/node_modules
/public
/src
/src-tauri

App
  tauri.rs - 1.0.0-beta-rc.0
build-type - bundle
CSP - default-src blob: data: filesystem: ws: http: https: 'unsafe-eval' 'unsafe-inline'

Done in 18.25s.
````

# - *edit (solution)* : update to visual studio 2019.

**Comment by MRakox** [comment]
Try to update Visual Studio Build Tools to the 2019 version
I had the same problem and it worked for me

**Reply by loicngr** [reply]
Thanks, just update to VS 2019 and works fine.

**Reply by alexprabhatbara** [reply]
I currently have VS 2022 Community Edition. I am still getting the same error. Please guide me.

**Reply by FabianLars** [reply]
You may have to uninstall any vs2015 / vs2017 installations you may have. Alternatively you can overwrite the global default or per terminal session but i don't know how tbh.
Also make sure you have selected all the components we show in the guide https://tauri.app/v1/guides/getting-started/prerequisites#1-microsoft-visual-studio-c-build-tools

And you may have to restart the system since we're talking about windows 🙄 

If that doesn't help please share the whole error message and the output of `tauri info`.

**Reply by alexprabhatbara** [reply]
Error message: thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 216, kind: Uncategorized, message: "This version of %1 is not compatible with the version of Windows you're running. Check your computer's system information and then contact the software publisher." }', src\interface\rust\desktop.rs:46:63
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

Output of tauri info:
Environment
  › OS: Windows 10.0.19045 X64
  › Webview2: 106.0.1370.52
  › MSVC:
      - Visual Studio Community 2022
  › Node.js: 19.0.0
  › npm: 8.19.2
  › pnpm: Not installed!
  › yarn: 1.22.19
  › rustup: 1.25.1
  › rustc: 1.64.0
  › cargo: 1.64.0
  › Rust toolchain: stable-x86_64-pc-windows-msvc

Packages
  › @tauri-apps/cli [NPM]: 1.1.1
  › @tauri-apps/api [NPM]: 1.1.0
  › tauri [RUST]: 1.1.1,
  › tauri-build [RUST]: 1.1.1,
  › tao [RUST]: 0.14.0,
  › wry [RUST]: 0.21.1,

App
  › build-type: bundle
  › CSP: unset
  › distDir: ../dist
  › devPath: http://localhost:1420/
  › framework: React

App directory structure
  ├─ node_modules
  ├─ public
  ├─ src
  └─ src-tauri

**Reply by FabianLars** [reply]
@alexprabhatbara Can you check that the C++ payload and the windows 10 sdk and c++ build tools (see screenshot in the guide) components are installed?
A screenshot would be appreciated so we can look into trying to reproduce it with the same versions you have selected.

**Comment by skygate2012** [comment]
In my case I've installed Visual Studio Build Tools 2019 but Windows 10 SDK was not selected.

**Comment by xmsz** [comment]
upgrade  Visual Studio, and reCreate project!!!
